backend/src/main/java/de/bund/digitalservice/ris/norms/application/service/ReleaseService.java

```diff
@@ -19,22 +31,149 @@ public class ReleaseService implements ReleaseAnnouncementUseCase {
     );
-    announcement.setReleasedByDocumentalistAt(Instant.now());
 
+    // find all norms to publish
+    Set<Norm> normsToPublish = new HashSet<>();
+    normsToPublish.add(normService.loadNorm(new LoadNormUseCase.Query(announcement.getEli())));
+
+    for (Norm norm : normsToPublish) {
+      var analysis = norm.getMeta().getAnalysis();
+      if (analysis.isEmpty()) {
+        break;
+      }
+
+      analysis
+        .get()
+        .getActiveModifications()
+        .stream()
+        .map(TextualMod::getDestinationHref)
+        .flatMap(Optional::stream)
+        .map(Href::getExpressionEli)
+        .flatMap(Optional::stream)
+        .map(eli -> normService.loadNorm(new LoadNormUseCase.Query(eli)))
+        .forEach(normsToPublish::add);
+    }
+
+    // Delete the files from a previous release of the same announcement if they are still queued for publishing.
+    // This is to prevent residual files from remaining if some time boundaries changed and this release will create
+    // different expressions.
+    normsToPublish.forEach(norm ->
+      deleteQueuedNormsPort.deleteQueuedForPublishNorms(
+        new DeleteQueuedNormsPort.Command(norm.getWorkEli())
+      )
+    );
+
+    // generate all future versions for all norms that will be published
+    Set<Norm> allVersionsOfAllNormsToPublish = new HashSet<>();
+    for (Norm norm : normsToPublish) {
+      // We need to publish the manifestation of the norm with the correct "nachfolgende-version-id", therefore we
+      // use the manifestation created from createNewExpression of the next time boundary if it exists
+      Norm latestNormExpression = createNewVersionOfNormService.createNewManifestation(norm);
+
+      var dates = norm
+        .getMeta()
+        .getAnalysis()
+        .stream()
+        .flatMap(analysis -> analysis.getPassiveModifications().stream())
+        .flatMap(textualMod -> textualMod.getForcePeriodEid().stream())
+        .map(norm::getStartDateForTemporalGroup)
+        .flatMap(Optional::stream)
+        .map(LocalDate::parse)
+        .distinct()
+        .sorted();
+
+      for (LocalDate date : dates.toList()) {
+        var result = createNewVersionOfNormService.createNewExpression(latestNormExpression, date);
+        allVersionsOfAllNormsToPublish.add(result.newManifestationOfOldExpression());
+
+        latestNormExpression =
+        timeMachineService.applyPassiveModifications(
+          new ApplyPassiveModificationsUseCase.Query(
+            result.newExpression(),
+            date.atStartOfDay(ZoneId.systemDefault()).toInstant()
+          )
+        );
+      }
+
+      allVersionsOfAllNormsToPublish.add(latestNormExpression);
+    }
+
+    allVersionsOfAllNormsToPublish.forEach(norm ->
+      norm.setPublishState(NormPublishState.QUEUED_FOR_PUBLISH)
+    );
+    allVersionsOfAllNormsToPublish.forEach(norm ->
+      updateOrSaveNormPort.updateOrSave(new UpdateOrSaveNormPort.Command(norm))
```

Comment:
You can also call the port in the same for each from before

Reply:
oh true will simplify that :)